rts/at/integration/src/main/java/org/jboss/narayana/rest/integration/RecoveryManager.java

```diff
@@ -93,12 +100,15 @@ public void removeParticipantInformation(final ParticipantInformation participan
         }
 
         final RecoveryStore recoveryStore = StoreManager.getRecoveryStore();
-        final Uid uid = new Uid(participantInformation.getId());
```

Comment:
A general question, how do we reach removeParticipantInformation if the participant was not property written to disk before. Do we need to differentiate on rollback vs other conditions?

Reply:
If I understand correctly what you ask about then the call to `removeParticipantInformation` is just called on these type of "actions" from the `ParticipantResource` (https://github.com/jbosstm/narayana/blob/5.10.4.Final/rts/at/integration/src/main/java/org/jboss/narayana/rest/integration/ParticipantResource.java#L276)
So any call on rollback, commit, forget may cause to call here. There is no differentiation from where the method is called from.

Reply:
It follows a little to the conversation @mmusgrov and you are having. It relates particularly to whether a failure to remove could be transient or not

Reply:
I probably don't understand the point. Failure to remove causes there is found the record in the next recovery round and the attempt to rollback/commit will be retried.

Reply:
I will resolve this as we are discussing something related in the parallel thread